src/test/test-file-descriptor.c

```diff
@@ -0,0 +1,59 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "file-descriptor.h"
+#include "macro-fundamental.h"
+#include "string-util.h"
+#include "tests.h"
+
+static void set_file_descriptor(int fd, const char* fdname, FileDescriptor* ret) {
+     *ret = (FileDescriptor) {
+        .fd = fd,
+        .fdname = fdname ? strdup(fdname) : NULL,
+    };
+}
+
+TEST(file_descriptor_validate) {
+        _cleanup_(file_descriptor_freep) FileDescriptor *fd_obj = NULL;
+        int r;
+
+        assert_se(fd_obj = new(FileDescriptor, 1));
```

Comment:
Please use the new assertion macros from `tests.h`, those give better error messages when an assertion fails.